numpy/polynomial/chebyshev.py

```diff
@@ -1886,6 +1887,59 @@ def chebroots(c):
     return r
 
 
+def chebinterp(func, deg, domain=None, args=()):
+    """Return the Chebyshev series instance that interpolates a given function
+    at the Chebyshev points of the first kind (mapped to the given domain).
+    The function is interpolated at the Chebyshev points of the first kind,
+    mapped onto the domain of the function being approximated.
+
+    Parameters
+    ----------
+    func : function
+        The function to be approximated. It must be a function of a single
+        variable of the form f(x,a,b,c...), where a,b,c... are extra arguments
+        that can be passed in the `args` parameter.
+    deg : int or 1-D array_like
+        Degree(s) of the fitting polynomials. If `deg` is a single
+        integer all terms up to and including the `deg`'th term are included in
+        the interpolation.
+    domain : [beg, end], optional
+        Domain to use for the returned series.
+    args : tuple, optional
+        Extra arguments to be used in the function call.
+
+    Returns
+    -------
+    new_series : series
+        A series that represents the polynomial interpolation
+        to the function.
+
+    Examples
+    --------
+    >>> from numpy import tanh, pi
+    >>> import numpy.polynomial.chebyshev as C
+    >>> C.chebinterp(lambda x: tanh(x)+0.5, 8, domain=[0, 2*np.pi])
+    array([ 1.30198814,  0.3540514 , -0.25209078,  0.14032589, -0.05706644,
+            0.01226391,  0.00367748, -0.00455335])
+
+    .. versionadded:: 1.14.0
+
+    """
+    window = chebdomain
```

Comment:
Should probably check that `deg` is a non-negative integer even though it will also be checked in `chebvander`.